### PR TITLE
Update basic_distributions syntax

### PR DESCRIPTION
**Pull request by andrjohns**
I'm going to start (slowly & sporadically) updating the syntax in these example models to modern Stan language, and remove deprecated syntax.

This PR updates the models in the ```basic_distributions``` folder. I've checked that all models parse under Stanc3 (with the exception of a [known issue](https://github.com/stan-dev/stanc3/issues/751)

### CODE REVIEW

**Comment by rok-cesnovar** [comment]
Nice! Thanks!

Do you check using rstan 2.26? cmdstan adds a _model suffix and rstan should as well to avoid the problem.

**Comment by andrjohns** [comment]
Ah good call, yep no errors under RStan 2.26

File: basic_distributions/binormal.stan [hunk 1]
```diff
@@ -1,13 +1,7 @@
...
-    vector[2] mu;
-
-    mu[1] <- 0.0;
-    mu[2] <- 0.0;
-    Sigma[1,1] <- 1.0;
-    Sigma[2,2] <- 1.0;
-    Sigma[1,2] <- 0.10;
-    Sigma[2,1] <- 0.10;
+    matrix[2,2] Sigma = [[1, 0.1],
+                         [0.1, 1]];
```
Comment (rok-cesnovar) [inline basic_distributions/binormal.stan:3]:
```suggestion
    matrix[2,2] Sigma = [[1, 0.1], [0.1, 1]];
```

File: basic_distributions/inv_wishart.stan [hunk 2]
```diff
@@ -1,8 +1,7 @@
 transformed data {
-  cov_matrix[3] S;
-  S[1, 1] <- 2; S[1, 2] <- 0; S[1, 3] <- 0; 
-  S[2, 2] <- 1; S[2, 1] <- 0; S[2, 3] <- 0;
-  S[3, 3] <- .5; S[3, 1] <- 0; S[3, 2] <- 0; 
+  cov_matrix[3] S = [[2, 0, 0],
+                     [0, 1, 0],
+                     [0, 0, 0.5]];
```
Comment (rok-cesnovar) [inline basic_distributions/inv_wishart.stan:4]:
```suggestion
  cov_matrix[3] S = [[2, 0, 0], [0, 1, 0], [0, 0, 0.5]];
```

File: basic_distributions/wishart.stan [hunk 3]
```diff
@@ -1,12 +1,6 @@
...
-  cov_matrix[2] S;
-
-  for (i in 1:2)
-    for (j in 1:2)
-      S[i,j] <- 0.0;
-
-  S[1,1] <- 2.0; 
-  S[2,2] <- 0.5;
+  cov_matrix[2] S = [[2, 0],
+                     [0, 0.5]];
```
Comment (rok-cesnovar) [inline basic_distributions/wishart.stan:3]:
```suggestion
  cov_matrix[2] S = [[2, 0], [0, 0.5]];
```

File: basic_distributions/wishart2.stan [hunk 4]
```diff
@@ -1,20 +1,20 @@
...
+  S[2,3] = 7.157993;
+  S[2,4] = -3.1129955;
+  S[3,3] = 11.676181;
+  S[3,4] = -3.5866852;
+  S[4,4] = 1.4482736;
 
   for (m in 1:4)
     for (n in 1:(m-1))
-      S[m,n] <- S[n,m];
+      S[m,n] = S[n,m];
```
Comment (rok-cesnovar) [inline basic_distributions/wishart2.stan:17]: If the above was filling the lower triangular parts we could use symmetrize_from_lower_tri but no sense complicating I guess.

File: basic_distributions/wishart2x2.stan [hunk 5]
```diff
@@ -10,32 +10,19 @@
...
-  cov_matrix[2] S;
-
-  for (i in 1:2)
-    for (j in 1:2)
-      S[i,j] <- 0.0;
-
-  S[1,1] <- 2.0;
-  S[2,2] <- 0.5;
+  cov_matrix[2] S = [[2, 0],
+                     [0, 0.5]];
```
Comment (rok-cesnovar) [inline basic_distributions/wishart2x2.stan:14]:
```suggestion
  cov_matrix[2] S = [[2, 0], [0, 0.5]];
```

File: basic_distributions/wishart2x2.stan [hunk 6]
```diff
@@ -10,32 +10,19 @@
...
-  cov <- rho * sd1 * sd2;
-
-  W[1,1] <- sd1 * sd1;
-  W[2,2] <- sd2 * sd2;
-  W[1,2] <- cov;
-  W[2,1] <- cov;
+  real rho = tanh(x);
+  real cov = rho * sd1 * sd2;
+  matrix[2,2] W = [[sd1^2, cov],
+                   [cov, sd2^2]];
```
Comment (rok-cesnovar) [inline basic_distributions/wishart2x2.stan:25]:
```suggestion
  matrix[2,2] W = [[sd1 ^ 2, cov], [cov, sd2 ^ 2]];
```

**Review by rok-cesnovar — COMMENTED** [review]
The auto-formatter from stanc3 suggests a bit different formatting in some places, though I guess its a matter of personal taste a bit so feel free to ignore.

Thanks for doing this!

**Comment by andrjohns** [comment]
Just for the example models I think I'd prefer to leave the matrix declarations in the matrix layout (rather than a one-liner), so that the ```[[],[]]``` declaration syntax is a bit more immediately intuitive for the unfamiliar. But will update the rest

**Comment by rok-cesnovar** [comment]
Yeah, its why I said these are optional. Let me know when to take another look.

**Comment by andrjohns** [comment]
Works for me! Ready for another look

**Review by rok-cesnovar — APPROVED** [review]
Thanks!

**Pull request merged**